miio/airpurifier_miot.py

```diff
@@ -0,0 +1,411 @@
+        return self.data["fan_level"]
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["mode"])
+
+    @property
+    def led(self) -> bool:
+        """Return True if LED is on."""
+        return self.data["led"]
+
+    @property
+    def led_brightness(self) -> Optional[LedBrightness]:
+        """Brightness of the LED."""
+        if self.data["led_brightness"] is not None:
+            try:
+                return LedBrightness(self.data["led_brightness"])
+            except ValueError:
+                return None
+
+        return None
+
+    @property
+    def buzzer(self) -> Optional[bool]:
+        """Return True if buzzer is on."""
+        if self.data["buzzer"] is not None:
+            return self.data["buzzer"]
+
+        return None
+
+    @property
+    def buzzer_volume(self) -> Optional[int]:
+        """Return buzzer volume."""
+        if self.data["buzzer_volume"] is not None:
+            return self.data["buzzer_volume"]
+
+        return None
+
+    @property
+    def child_lock(self) -> bool:
+        """Return True if child lock is on."""
+        return self.data["child_lock"]
+
+    @property
+    def favorite_level(self) -> int:
+        """Return favorite level, which is used if the mode is ``favorite``."""
+        # Favorite level used when the mode is `favorite`.
+        return self.data["favorite_level"]
+
+    @property
+    def filter_life_remaining(self) -> int:
+        """Time until the filter should be changed."""
+        return self.data["filter_life_remaining"]
+
+    @property
+    def filter_hours_used(self) -> int:
+        """How long the filter has been in use."""
+        return self.data["filter_hours_used"]
+
+    @property
+    def use_time(self) -> int:
+        """How long the device has been active in seconds."""
+        return self.data["use_time"]
+
+    @property
+    def purify_volume(self) -> int:
+        """The volume of purified air in cubic meter."""
+        return self.data["purify_volume"]
+
+    @property
+    def motor_speed(self) -> int:
+        """Speed of the motor."""
+        return self.data["motor_speed"]
+
+    @property
+    def filter_rfid_product_id(self) -> Optional[str]:
+        """RFID product ID of installed filter."""
+        return self.data["filter_rfid_product_id"]
+
+    @property
+    def filter_rfid_tag(self) -> Optional[str]:
+        """RFID tag ID of installed filter."""
+        return self.data["filter_rfid_tag"]
+
+    @property
+    def filter_type(self) -> Optional[FilterType]:
+        """Type of installed filter."""
+        if self.filter_rfid_tag is None:
+            return None
+        if self.filter_rfid_tag == "0:0:0:0:0:0:0":
+            return FilterType.Unknown
+        if self.filter_rfid_product_id is None:
+            return FilterType.Regular
+        return self.filter_type_util.determine_filter_type(self.filter_rfid_product_id)
```

Comment:
Would it make sense to pass both tag & product_id to the `determine_filter_type` and make all the above checks there?

Reply:
Makes sense. Will do